mail_parent_recipient/models/mail_template.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models, api
+
+
+class MailTemplate(models.Model):
+
+    _inherit = 'mail.template'
+
+    @api.multi
+    def generate_recipients(self, results, res_ids):
+        """
+        Modify the code that set the default recipient in an email to the first
+        partner having an email set following the parents tree.
+        If no parent with an email is found,
+        it reverts to the popup asking to set an email for the current partner
+        """
+        results = super(MailTemplate, self).generate_recipients(
+            results,
+            res_ids
+        )
+        for res_id, values in results.iteritems():
```

Comment:
nice to have maybe: I ctx flag to disable this behavior?

Reply:
As stated by @chienandalu and @pedrobaeza you need to be nice to others modules' tests.

Use the config['test_enabled'] condition for that.